extensions/amp-form/0.1/amp-form.js

```diff
@@ -109,6 +116,26 @@ export class AmpForm {
     }
     this.form_.classList.add(`amp-form-${state}`);
   }
+
+  /**
+   * @param {string} state
+   * @param {!Object} data
+   * @private
+   */
+  renderTemplate_(state, data) {
+    const container = this.form_.querySelector(`[${state}]`);
+    if (container) {
+      const previousRender = container.querySelector(`.${state}-message`);
```

Comment:
How about removing this element during the submit? (It's also fix transitioning from error -> success and success -> error that this has).

From a UI perspective, if the POST takes a few seconds to finish, I could be staring at an error state thinking my input is bad.